src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3424,7 +3410,7 @@ private static TypeWithState ApplyUnconditionalAnnotations(TypeWithState typeWit
         {
             if ((annotations & FlowAnalysisAnnotations.MaybeNull) == FlowAnalysisAnnotations.MaybeNull)
             {
-                return TypeWithState.Create(typeWithState.Type, NullableFlowState.MaybeNull);
+                return TypeWithState.Create(typeWithState.Type, NullableFlowState.MaybeDefault);
```

Comment:
>MaybeDefault [](start = 82, length = 12)

Isn't `MaybeDefault` only supposed to be used for unconstrained type parameters?

Reply:
If the type is not an unconstrained type parameter, `TypeWithState.Create()` will use `MaybeNull` instead.

---
In reply to: [347686560](https://github.com/dotnet/roslyn/pull/39778#discussion_r347686560) [](ancestors = 347686560)